lib/hexpm/repository/package_reports.ex

```diff
@@ -32,25 +40,110 @@ defmodule Hexpm.Repository.PackageReports do
     |> Repo.one()
     |> PackageReport.change_state(%{"state" => "accepted"})
     |> Repo.update()
+
+    report = PackageReport.get(report_id)
+
+    report =
+      PackageReport.get(report_id)
+      |> Repo.one()
+
+    email_user_about_state_change(report, report.author)
+
+    Enum.each(
+      Owners.all(report.package),
+      fn owner ->
+        owner = Hexpm.Repo.preload(owner, user: [])
+        email_user_about_state_change(report, owner.user)
+      end
+    )
+
+    Enum.each(
+      Users.get_by_role("moderator"),
+      fn user ->
+        email_user_about_state_change(report, user)
+      end
+    )
   end
 
   def reject(report_id, comment) do
     PackageReport.get(report_id)
     |> Repo.one()
     |> PackageReport.change_state(%{"state" => "rejected"})
     |> Repo.update()
+
+    report =
+      PackageReport.get(report_id)
+      |> Repo.one()
+
+    email_user_about_state_change(report, report.author)
+
+    Enum.each(
+      Owners.all(report.package),
+      fn owner ->
+        owner = Hexpm.Repo.preload(owner, user: [])
+        email_user_about_state_change(report, owner.user)
+      end
+    )
+
+    Enum.each(
+      Users.get_by_role("moderator"),
+      fn user ->
+        email_user_about_state_change(report, user)
+      end
+    )
   end
 
   def solve(report_id, comment) do
-    PackageReport.get(report_id)
+    report = PackageReport.get(report_id)
+
+    report
     |> Repo.one()
     |> PackageReport.change_state(%{"state" => "solved"})
     |> Repo.update()
+
+    report =
+      PackageReport.get(report_id)
+      |> Repo.one()
+
+    email_user_about_state_change(report, report.author)
+
+    Enum.each(
+      Owners.all(report.package),
+      fn owner ->
+        owner = Hexpm.Repo.preload(owner, user: [])
+        email_user_about_state_change(report, owner.user)
+      end
+    )
+
+    Enum.each(
+      Users.get_by_role("moderator"),
+      fn user ->
+        email_user_about_state_change(report, user)
+      end
+    )
```

Comment:
It will be less code if you do something like: `[report.author] ++ Owners.all(report.package) ++ Users.get_by_role("moderator")` and then run the `Enum.each`

Reply:
I used a separated Each for owners since I think I need to take the user from each owner.
`Enum.each(
      Owners.all(comment.report.package, user: []),
        &email_user_about_new_comment(comment, **$1.user**)
    )`

Reply:
Ah, good point. You can get the users for owners by doing something like: `Enum.map(Owners.all(report.package), & &1.user)`.